whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapRedirectTestIntegration.java

```diff
@@ -114,6 +114,44 @@ public void autnum_no_redirect_property() {
         });
     }
 
+    // domain
+
+    @Test
+    public void redirect_domain_query() {
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "domain/7.0.193.in-addr.arpa"))
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(String.class);
+            fail();
+        } catch (final RedirectionException e) {
+            assertThat(e.getResponse().getHeaders().getFirst("Location").toString(), is("https://rdap.one.net/domain/7.0.193.in-addr.arpa"));
+        }
+    }
+
+    @Test
+    public void redirect_ipv6_domain_query() {
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "domain/0.7.3.0.c.7.6.0.1.0.0.2.ip6.arpa"))
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(String.class);
+            fail();
+        } catch (final RedirectionException e) {
+            assertThat(e.getResponse().getHeaders().getFirst("Location").toString(), is("https://rdap.one.net/domain/0.7.3.0.c.7.6.0.1.0.0.2.ip6.arpa"));
+        }
+    }
+
+    @Test
+    public void domain_outside_range() {
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "domain/0.0.192.in-addr.arpa"))
```

Comment:
maybe to avoid this situations (empty catch) we could use 
https://www.baeldung.com/integration-testing-in-spring
use MockMvc instead of WebTarget

Reply:
Ah, I should use `Assertions.assertThrows()` it's built-in to Junit 5

Reply:
I preferred `@Test(expected = XXX.class)` in JUnit 4 but it's dropped